integrations/ragas/src/haystack_integrations/components/evaluators/ragas/metrics.py

```diff
@@ -170,12 +157,51 @@ def new(
         return cls(
             metric=metric,
             backend=backend,
+            input_validator=input_validator,
             input_parameters=input_parameters,
             input_converter=input_converter,
+            output_converter=output_converter if output_converter is not None else OutputConverters.default,
             init_parameters=init_parameters,
         )
 
 
+class InputValidators:
+    """
+    Validates for input parameters.
+
+    The signature of the converter functions serves as the ground-truth of the
+    expected input parameters of a given metric. They are also responsible for validating
+    the input parameters and converting them to the format expected by Ragas.
+    """
+
+    @staticmethod
+    def validate_empty_metric_parameters(metric: RagasMetric, metric_params: Dict[str, Any]) -> None:
+        if metric_params:
+            msg = f"Unexpected init parameters '{metric_params}' for metric '{metric}'."
+            raise ValueError(msg)
+
+    @staticmethod
+    def validate_aspect_critique_parameters(metric: RagasMetric, metric_params: Dict[str, Any]) -> None:
```

Comment:
Looks like aspect critique always requires all of its metric params? If that's correct, we can simplify this into a `all(k in metric_params for k in expected_metric_params.keys())` generic function that takes `metric, expected_params, received_params` as inputs.

Reply:
I change the logic to use one validator function by default, which takes `metric`, `expected_params`, `received_params` as inputs.